aten/src/ATen/mps/MPSDevice.h

```diff
@@ -0,0 +1,58 @@
+//  Copyright © 2022 Apple Inc.
+
+#pragma once
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+#include <ATen/ATen.h>
+
+
+#ifdef __OBJC__
+#include <Foundation/Foundation.h>
+#include <Metal/Metal.h>
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+typedef id<MTLDevice> MTLDevice_t;
+#else
+typedef void* MTLDevice;
+typedef void* MTLDevice_t;
+#endif
+
+using namespace std;
+
+namespace at {
+namespace mps {
+
+// MPSDevice is a singleton class that returns the default device
```

Comment:
This seems to suggest that there is only one valid MPS device. Is this the appropriate assumption to make here?